src/command.lisp

```diff
@@ -1171,6 +1171,136 @@ _~~A() {
           (format stream "~A~%" line)))
       (format stream "~%"))))
 
+(defmethod print-documentation ((kind (eql :mandoc)) (top-level command) stream &key (wrap-at 80))
+  "Generates section 1 man pages in the mdoc(7) format.  Documentation
+available at https://man.openbsd.org/mdoc.7"
+  ;; FIXME: Other man sections are "context, implementation notes,
+  ;; return values, files, exit status, diagnostics, errors, see also,
+  ;; standards, history, caveats, bugs, and security considerations";
+  ;; not sure how we'd go about generating them, but it would be good
+  ;; to do so.
+  (labels ((wrap-paragraph (par)
+             (split-sequence #\Newline (bobbin:wrap par wrap-at)))
+           (timestamp ()
+             (let ((now (multiple-value-list (decode-universal-time (get-universal-time)))))
+               (format NIL "~A-~A-~A" (nth 5 now) (nth 4 now) (nth 3 now)))))
+    (format stream ".Dd ~A~%~
+                  .Dt ~:@(~A~) 1~%~
+                  .Os~%~
+                  .Sh NAME~%~
+                  .Nm ~:*~A~%~
+                  .Nd ~A~%"
+            (timestamp)
+            (command-full-name top-level)
+            (command-description top-level))
+
+    (format stream ".Sh SYNOPSIS~%")
+    (with-command-tree (node top-level)
+      (initialize-command node)
+      (format stream ".Nm ~A~%" (command-full-name node))
+      (loop :for opt :in (command-options node)
+            :unless (option-hidden-p opt)
+              :do (format stream ".Op Fl~:[~:; ~:*~A,~]~
+                                 ~:[~:; -~:*~A~]~
+                                 ~:[~:; Ar ~:*~A~]~%"
+                          (option-short-name opt)
+                          (option-long-name opt)
+                          (option-parameter opt)))
+      ;; FIXME: As it stands there is not really any good way to
+      ;; display the command's arguments; the closest thing is
+      ;; `command-usage' with the huge downside that clingon wants you
+      ;; to specify arguments in there too, which clashes really badly
+      ;; with the rest of the synopsis.  Need to ask advice on how
+      ;; best to approach this.
+      (when (command-usage node)
+        (format stream ".Op Ar ~A~%" (command-usage node))))
+
+    ;; FIXME: It would be good to group options by category, but I'm
```

Comment:
Do you mean this one?

* https://github.com/dnaeon/clingon/blob/master/src/command.lisp#L963

Reply:
Yes.  Problem is, from my understanding of the mandoc format, the group names would come out looking like subcommands which doesn't seem ideal.  There's probably some workaround though, I need to read the man page further.